crates/proto/src/h2/h2_client_stream.rs

```diff
@@ -571,6 +604,12 @@ impl Future for HttpsClientResponse {
     }
 }
 
+/// Trait that can be used to implement a custom type to return dynamic headers to add to a DoH request
+pub trait AddHeaders: Send + Sync + 'static {
+    /// Get a set of headers to add to the query
+    fn get_headers(&self) -> Vec<(HeaderName, HeaderValue)>;
```

Comment:
WDYT about naming this just `headers()`? I think the `get_` prefix is probably unnecessary.